torch/testing/_internal/common_utils.py

```diff
@@ -1820,6 +1820,32 @@ def __init__(self, actual, expected, *, rtol_override=0.0, atol_override=0.0, **
         self.rtol = max(self.rtol, rtol_override)
         self.atol = max(self.atol, atol_override)
 
+        # This is a slow and ugly hack to allow the comparison of hybrid sparse CSR tensors with strided ones. If
+        # `check_layout=False` (default), the tensors will be converted to strided by calling `.to_dense()` on them.
+        # However, this is not yet supported for hybrid sparse CSR and thus we need to do it manually for now.
+        # FIXME: Remove this as soon as `.to_dense` is supported for hybrid sparse CSR tensors
+        if not self.check_layout:
+            self.actual, self.expected = self._handle_hybrid_sparse_csr(self.actual, self.expected)
+
+    def _handle_hybrid_sparse_csr(self, actual, expected):
+        if not ((actual.layout is torch.sparse_csr) ^ (expected.layout is torch.sparse_csr)):
+            return actual, expected
```

Comment:
To support other sparse compressed layouts:
```suggestion
        compressed_sparse_layouts = {torch.sparse_csr, torch.sparse_csc, torch.sparse_bsr, torch.sparse_bsc}
        if not ((actual.layout in compressed_sparse_layouts) ^ (expected.layout in compressed_sparse_layouts)):
            return actual, expected
```